contrib/mesos/pkg/service/endpoints_controller.go

```diff
@@ -325,7 +324,6 @@ func (e *endpointController) syncService(key string) {
 			containerPortAnnotations[fmt.Sprintf(meta.ContainerPortKeyFormat, portProto, pod.Status.HostIP, portNum)] = strconv.Itoa(containerPort)
 		}
 	}
-	subsets = endpoints.RepackSubsets(subsets)
```

Comment:
In general I agree with caution about those changes. But here it's not worse than what we do already:
* the host ports (instead of container ports) lead to multiple subsets for each host already
* the only thing we do here in addition (with leaving out the repack call), is to avoid combining all ports of a host into one subset.

Some things to think about:
* It could be that the repacking is also done in the apiserver. Then leaving out this call does not help at all.
* We still don't know which constraints are really needed by the actual kube-proxy code. Before accepting any solution we have to understand the problem better.